### PR TITLE
mit-scheme: 10.1.10 -> 11.2

### PR DESCRIPTION
**Pull request by leungbk**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`) (**I don't have an ARM64 processor, so I would appreciate it if someone who does could try this PR**)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 133017` at 36b1b165 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>mitscheme</li>
    <li>mitschemeX11</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/compilers/mit-scheme/default.nix:90:5:
```
   |
90 |     maintainers = [ ];
   |     ^
```
</li>
  </ul>
</details>

File: pkgs/development/compilers/mit-scheme/default.nix [hunk 1]
```diff
@@ -16,49 +17,55 @@ stdenv.mkDerivation {
...
+      url = "mirror://gnu/mit-scheme/stable.pkg/${version}/mit-scheme-${version}-aarch64le.tar.gz";
+      sha256 = "11maixldk20wqb5js5p4imq221zz9nf27649v9pqkdf8fv7rnrs9";
   } else fetchurl {
       url = "mirror://gnu/mit-scheme/stable.pkg/${version}/mit-scheme-${version}-x86-64.tar.gz";
-      sha256 = "1rljv6iddrbssm91c0nn08myj92af36hkix88cc6qwq38xsxs52g";
+      sha256 = "17822hs9y07vcviv2af17p3va7qh79dird49nj50bwi9rz64ia3w";
     };
 
-  buildInputs = if enableX11 then [xlibsWrapper] else [];
+  buildInputs = [ ncurses ] ++ (if enableX11 then [xlibsWrapper] else []);
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/mit-scheme/default.nix:29]:
```suggestion
  buildInputs = [ ncurses ] ++ lib.optional enableX11 xlibsWrapper;
```

File: pkgs/development/compilers/mit-scheme/default.nix [hunk 2]
```diff
@@ -16,49 +17,55 @@ stdenv.mkDerivation {
...
+  '';
 
-  fixupPhase =
+  postFixup =
     '' wrapProgram $out/bin/mit-scheme${arch} --set MITSCHEME_LIBRARY_PATH \
          $out/lib/mit-scheme${arch}
     '';
 
-  nativeBuildInputs = [ makeWrapper gnum4 texinfo texLive automake ];
+  nativeBuildInputs = [ makeWrapper gnum4 texinfo texLive automake ghostscript autoconf libtool];
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/mit-scheme/default.nix:68]:
```suggestion
  nativeBuildInputs = [ makeWrapper gnum4 texinfo texLive automake ghostscript autoconf libtool ];
```

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -11920,14 +11920,11 @@ in
   mint = callPackage ../development/compilers/mint { };
 
   mitscheme = callPackage ../development/compilers/mit-scheme {
-    texLive = texlive.combine { inherit (texlive) scheme-small; };
-    texinfo = texinfo5;
+    texLive = texlive.combine { inherit (texlive) scheme-small epsf texinfo; };
     xlibsWrapper = null;
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:11924]:
```suggestion
```

File: pkgs/development/compilers/mit-scheme/default.nix [hunk 4]
```diff
@@ -1,12 +1,13 @@
...
 let
-  version = "10.1.10";
-  bootstrapFromC = ! (stdenv.isi686 || stdenv.isx86_64);
+  version = "11.2";
+  bootstrapFromC = ! ((stdenv.isLinux && stdenv.isAarch64) || stdenv.isx86_64);
 
-  arch = if      stdenv.isi686   then "-i386"
-         else                         "-x86-64";
+  arch = if stdenv.isLinux && stdenv.isAarch64 then "-aarch64le"
+         else                                       "-x86-64";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/mit-scheme/default.nix:10]:
```suggestion
  arch = if stdenv.isLinux && stdenv.isAarch64 then
    "-aarch64le"
   else
     "-x86-64";
```

File: pkgs/development/compilers/mit-scheme/default.nix [hunk 5]
```diff
@@ -1,12 +1,13 @@
 { fetchurl, lib, stdenv, makeWrapper, gnum4, texinfo, texLive, automake,
+  autoconf, libtool, ghostscript, ncurses,
   enableX11 ? false, xlibsWrapper ? null }:
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/mit-scheme/default.nix:3]:
```suggestion
  enableX11 ? false, xlibsWrapper }:
```

**Pull request merged**